src/Main.elm

```diff
@@ -222,53 +248,90 @@ view model =
 
 pageContent : Model -> Html Msg
 pageContent model =
-    let
-        daysCompleted =
-            timesHabitWasCompleted model.habit model.habitLog
+    div []
+        [ habitScreen model
+        , progressBar model
+        , editHabitModal model
+        ]
 
-        daysCompletedText =
-            String.fromInt daysCompleted ++ " / 21"
-
-        habitCompleteButton =
-            if completedToday model then
-                div []
-                    [ div
-                        [ class <| primaryButton ++ " text-gray-800"
-                        , onClick RemoveLastHabitEntry
-                        ]
-                        [ Icons.undo purple 36 ]
-                    ]
 
-            else
-                button
-                    [ class <| primaryButton ++ " bg-purple-700 hover:bg-purple-800", onClick CompleteHabit ]
-                    [ Icons.check white 36 ]
+daysCompleted : Model -> Int
+daysCompleted model =
+    timesHabitWasCompleted model.habit model.habitLog
+
+
+habitScreen : Model -> Html Msg
+habitScreen model =
+    let
+        daysCompletedText =
+            String.fromInt (daysCompleted model) ++ " / 21"
     in
-    div []
-        [ div [ class "grid h-screen px-4 py-12" ]
-            [ div [ class "flex flex-col justify-end" ]
-                [ h1
-                    [ classList
-                        [ ( header ++ " overflow-y-scroll max-h-64 mb-10 break-anywhere", True )
-                        , ( "line-through", completedToday model )
-                        ]
-                    , onClick ToggleEditHabit
-                    ]
-                    [ text model.habit ]
-                , div [ class daysText ] [ text daysCompletedText ]
-                ]
-            , div [ class "flex items-end justify-center" ] [ habitCompleteButton ]
+    div [ class "fixed grid h-screen w-screen px-4 py-12 z-10" ]
+        [ div [ class "flex flex-col justify-end" ]
+            [ habitTextView model
+            , div [ class daysText ] [ text daysCompletedText ]
             ]
-        , editHabitModal model
+        , div [ class "flex items-end justify-center" ] [ habitCompleteButton model ]
         , completedTodayText model
         ]
 
 
+progressBar : Model -> Html Msg
+progressBar model =
+    let
+        totalDays =
```

Comment:
I wonder if we should pull this out into a global? or maybe a part of the model? I know we are committed to the 21 days, but it feels like a bit of a magic number here.

Reply:
Sounds good!